apis/apps/v1alpha1/opsrequest_types.go

```diff
@@ -311,12 +311,50 @@ type Upgrade struct {
 	//
 	// This field is deprecated since v0.9 because ClusterVersion is deprecated.
 	//
-	// +kubebuilder:validation:Required
 	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.9.0"
-	ClusterVersionRef string `json:"clusterVersionRef"`
+	ClusterVersionRef string `json:"clusterVersionRef,omitempty"`
+
+	// Lists components to be upgrade based on desired ComponentDefinition and ServiceVersion.
+	// Support the following user scenarios:
+	// 1. empty componentDefinitionName + non-empty serviceVersion:
+	//    Upgrade to the specified version and latest compatible ComponentDefinition.
+	//    But the prerequisite for this scenario is that cluster.spec.clusterDefinition must not be empty.
+	// 2. non-empty componentDefinitionName + empty serviceVersion:
+	//    If ComponentVersion of the ComponentDefinition is not defined,
+	//    the images specified in ComponentDefinition will be used.
+	//    Otherwise, the latest available version of the ComponentDefinition will be used.
```

Comment:
should be:

If it doesn't exist any ComponentVersion CR that provides the releases compatibility information about this ComponentDefinition CR, then the images specified in the ComponentDefinition CR will be used as a fallback.
Otherwise, the images from the release of the latest version will be used instead.